pds_doi_core/cmd/pds_doi_cmd.py

```diff
@@ -0,0 +1,242 @@
+#!/bin/python
+#
+#  Copyright 2020, by the California Institute of Technology.  ALL RIGHTS
+#  RESERVED. United States Government Sponsorship acknowledged. Any commercial
+#  use must be negotiated with the Office of Technology Transfer at the
+#  California Institute of Technology.
+#
+# ------------------------------
+
+from lxml import etree
+
+from pds_doi_core.util.cmd_parser import create_cmd_parser
+from pds_doi_core.util.const import *
+
+from pds_doi_core.util.config_parser import DOIConfigUtil
+from pds_doi_core.util.general_util import DOIGeneralUtil, get_logger
+from pds_doi_core.input.input_util import DOIInputUtil
+from pds_doi_core.input.pds4_util import DOIPDS4LabelUtil
+from pds_doi_core.input.validation_util import DOIValidatorUtil
+from pds_doi_core.cmd.DOIWebClient import DOIWebClient
+
+# Get the common logger and set the level for this file.
+import logging
+logger = get_logger('pds_doi_core.cmd.pds_doi_cmd')
+#logger.setLevel(logging.INFO)  # Comment this line once happy with the level of logging set in get_logger() function.
+#logger.setLevel(logging.DEBUG)  # Comment this line once happy with the level of logging set in get_logger() function.
+# Note that the get_logger() function may already set the level higher (e.g. DEBUG).  Here, we may reset
+# to INFO if we don't want debug statements.
+
+class DOICoreServices:
+    m_doiConfigUtil = DOIConfigUtil()
+    m_doiGeneralUtil = DOIGeneralUtil()
+    m_doiInputUtil = DOIInputUtil()
+    m_doiPDS4LabelUtil = DOIPDS4LabelUtil()
+    m_doiValidatorUtil = DOIValidatorUtil()
+    m_doiWebClient = DOIWebClient()
+
+    def reserve_doi_label(self, target_url, publisher_value, contributor_value):
+        """
+        Function receives a URI containing either XML, SXLS or CSV and create one or many labels to disk and submit these label(s) to OSTI.
+        :param target_url:
+        :param publisher_value:
+        :param contributor_value:
+        :return:
+        """
+
+        action_type = 'reserve_osti_label'
+        o_doi_label = 'invalid action type:action_type ' + action_type
+
+        logger.debug(f"target_url,action_type {target_url} {action_type}")
+
+        file_is_parsed_flag = False
+
+        if target_url.endswith('.xml'):
+            o_doi_label = self.m_doiPDS4LabelUtil.parse_pds4_label_via_uri(target_url, publisher_value, contributor_value)
+
+            file_is_parsed_flag = True
```

Comment:
this variable is not useful if we use elif statement

Reply:
Variable has been removed.

Reply:
Variable has been removed.